includes/class-wc-stripe-account.php

```diff
@@ -0,0 +1,160 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Stripe_Account class.
+ *
+ * Communicates with Stripe API.
+ */
+class WC_Stripe_Account {
+
+	const ACCOUNT_OPTION = 'wcstripe_account_data';
+	const ACCOUNT_API    = 'account';
+
+	public function __construct() {
+		self::cache_account();
+	}
+
+	/**
+	 * Gets and caches the data for the account connected to this site.
+	 *
+	 * @return array Account data or empty if failed to retrieve account data.
+	 */
+	public static function get_cached_account_data() {
+		if ( ! woocommerce_gateway_stripe()->connect->is_connected() ) {
+			return [];
+		}
+
+		$account = self::read_account_from_cache();
+
+		if ( ! empty( $account ) ) {
+			return $account;
+		}
+
+		$account = self::cache_account();
+		return $account;
+	}
+
+	/**
+	 * Read the account from the WP option we cache it in.
+	 *
+	 * @return array
+	 */
+	private static function read_account_from_cache() {
+		$account_cache = get_option( self::ACCOUNT_OPTION );
+
+		if ( false === $account_cache || ! isset( $account_cache['account'] ) || ! isset( $account_cache['expires'] ) ) {
+			// No option found or the data isn't in the shape we expect.
+			return [];
+		}
+
+		// Set $account to empty if the cache has expired, triggering another fetch.
+		if ( $account_cache['expires'] < time() ) {
+			return [];
+		}
+
+		// We have fresh account data in the cache, so return it.
+		return $account_cache['account'];
+	}
+
+	/**
+	 * Caches account data for a period of time.
+	 */
+	private static function cache_account() {
+		$expiration = 2 * HOUR_IN_SECONDS;
+
+		try {
+			$account = WC_Stripe_API::request(
+				[],
+				self::ACCOUNT_API,
+				'GET'
+			);
+		} catch ( WC_Stripe_Exception $e ) {
+			return [];
+		}
+
+		// Add the account data and expiry time to the array we're caching.
+		$account_cache            = [];
+		$account_cache['account'] = $account;
+		$account_cache['expires'] = time() + $expiration;
+
+		// Create or update the account option cache.
+		if ( false === get_option( self::ACCOUNT_OPTION ) ) {
+			add_option( self::ACCOUNT_OPTION, $account_cache, '', 'no' );
+		} else {
+			update_option( self::ACCOUNT_OPTION, $account_cache, 'no' );
+		}
```

Comment:
Why not just use `update_option`? Is it necessary to check if the option was set before?

BTW, WordPress has a similar mechanism for this "caching" behavior - transients: https://developer.wordpress.org/apis/handbook/transients/#using-transients
You can set a cache and you'll be able to cache the response for a certain amount of time. When the cache expires the data is removed, tho.

Reply:
Thanks for the link. Using transient now. :smile: